src/java/org/apache/cassandra/concurrent/Stage.java

```diff
@@ -124,10 +129,20 @@ public static Stage fromPoolName(String stageName)
     // Convenience functions to execute on this stage
     public void execute(Runnable command) { executor().execute(command); }
     public void execute(Runnable command, ExecutorLocals locals) { executor().execute(command, locals); }
-    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); };
+    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); }
     public <T> Future<T> submit(Callable<T> task) { return executor().submit(task); }
     public Future<?> submit(Runnable task) { return executor().submit(task); }
     public <T> Future<T> submit(Runnable task, T result) { return executor().submit(task, result); }
+    public Runnable submitIfNotShutdown(Runnable task)
+    {
+        if (MIGRATION.executor().isShutdown() || MIGRATION.executor().isTerminated())
+        {
+            logger.info("Skipped scheduled pulling schema from other nodes: the node is decommissioning.");
+            return Runnables.doNothing();
+        }
+        else
+            return ()->MIGRATION.submit(task);
```

Comment:
...and I think what @yifan-c is getting at is that the submission here should be synchronous, not wrapped in a runnable...i.e. the submission should either not happen or happen immediately after we hit the `else` here. (We ignore the return value of `submitIfNotShutdown()` anyway?)

Reply:
Yes, I confirmed with him and now it clicked with your other comment for code repetition. 
Pushing a commit, thanks :-)